aodnfetcher/fetcherlib.py

```diff
@@ -619,6 +621,49 @@ def _get_matching_builds(self):
         return sorted_keys
 
 
+class PrefixS3Fetcher(BaseResolvingS3Fetcher):
+    """Fetch the latest artifact under an s3 prefix, using Etag header to provide identifier for cache validation """
+
+    def __init__(self, parsed_url, local_file_hint=None, authenticated=False):
+        super(PrefixS3Fetcher, self).__init__(parsed_url, local_file_hint, authenticated)
+
+        self.prefix = parsed_url.path.lstrip('/')
+
+        self._all_builds = None
+        self._filename_pattern = None
+
+    @property
+    def all_builds(self):
+        if self._all_builds is None:
+            self._all_builds = [k for k in _paginate(self.s3_client.list_objects_v2, Bucket=self.bucket,
+                                                     Prefix=self.prefix)]
+        return self._all_builds
+
+    @property
+    def filename_pattern(self):
+        if self._filename_pattern is None:
+            self._filename_pattern = self.get_value_from_query_string('pattern', r'^.*\.war$')
+        return self._filename_pattern
+
+    def _get_key(self):
+        if not self.all_builds:
+            raise KeyResolutionError('NO_RESULTS',
+                                     "prefix '{s.prefix}' was invalid or returned no artifacts".format(s=self))
+
+        try:
+            return self._get_latest_matching_key()
+        except IndexError:
+            raise KeyResolutionError('NO_MATCHING_KEYS',
+                                     "no keys found for '{s.prefix}' matching '{s.filename_pattern}'".format(
+                                         s=self))
+
+    def _get_latest_matching_key(self):
+        matching_keys = (a for a in self.all_builds if
+                         re.match(self.filename_pattern, a['Key']))
+        sorted_by_date = sorted(matching_keys, key=lambda p: p['LastModified'])
```

Comment:
I might be more intuitive to return the latest *version* rather than the last modified?

If I was looking in a prefix with:
```
mysoftware-1.2.1.war
mysoftware-1.2.2.war
```

My default expectation would be that it would return `mysoftware-1.2.2.war`, regardless of the LastModified date.

https://stackoverflow.com/questions/11887762/how-do-i-compare-version-numbers-in-python

Thinking on from that, it might be desirable in *some* cases to sort by date, and some cases sort by version. This could be achieved in the same way as the `pattern` URL parameter, which uses get_value_from_query_string.

e.g.
```
s3prefix://bucket/key  #would use the default sortmethod, whichever that was
s3prefix://bucket/key?sortmethod=version # explicitly sort by version
s3prefix://bucket/key?sortmethod=newest  # explicitly sort by newest
```

Reply:
Implemented in latest commit